Assets/Scripts/ThirdPersonPlayer.cs

```diff
@@ -60,36 +77,71 @@ public void OnTriggerEnter(Collider collision)
     private void Update()
     {
         stardustCount.text = "Stardust: " + stardust;
-
-        Debug.Log(stardust);
     }
 
     private void FixedUpdate()
     {
+    
+        if (_levelCleared) return;
+        
         /*
         var xAxisInput = Input.GetAxisRaw("Horizontal");
         var yAxisInput = Input.GetAxisRaw("Vertical");
          */
 
         //InputAction replaces "Input.GetAxis("Example")" and calls function
         //movementInput = inputAction.Player.Move.ReadValue<Vector2>();
+        
+        if (_mapActive) return;
+
+        // when on slingshot, make the player move towards the target 
+        if (_onSlingShot)
+        {
+            print("|||||||||| third person player --- slngshot ||||||||||" + _starToGo.name);
+            
+            // sling shot the player to the designated star starToGo
+            var desiredPosition = _starToGo.position + disFromGoalStar;
+            //transform.position = Vector3.MoveTowards(transform.position, desiredPosition, speed * 100 * Time.fixedDeltaTime);
+            transform.position = Vector3.Lerp(transform.position, desiredPosition, speed * 2 * Time.fixedDeltaTime);
+
+            // TODO: disable all the figure when slingshot
```

Comment:
I hope you plan to implement this TODO yourself, since I have no idea what this TODO is supposed to be.